### ISSUE TITLE
monkey_patch() error when used with nosetests

### ISSUE THREAD

**Issue by j-o-d-o** [eventlet/eventlet#464]
OS: Windows 10
Python: 3.5.3
eventlet: 0.22.0

I am using eventlet for [Flask-SocketIO](https://github.com/miguelgrinberg/Flask-SocketIO). When testing my API endpoints, this error occures: (only when using nosetests, using unittest works, but I have to use nosetests here...)

```
======================================================================
ERROR: Failure: ReferenceError (weakly-referenced object no longer exists)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "....\env\lib\site-packages\nose\failure.py", line 39, in runTest
    raise self.exc_val.with_traceback(self.tb)
  File "....\env\lib\site-packages\nose\loader.py", line 418, in loadTestsFromName
    addr.filename, addr.module)
  File "....\env\lib\site-packages\nose\importer.py", line 47, in importFromPath
    return self.importFromDir(dir_path, fqname)
  File "....\env\lib\site-packages\nose\importer.py", line 94, in importFromDir
    mod = load_module(part_fqname, fh, filename, desc)
  File "C:\Anaconda3\lib\imp.py", line 235, in load_module
    return load_source(name, filename, file)
  File "C:\Anaconda3\lib\imp.py", line 172, in load_source
    module = _load(spec)
  File "<frozen importlib._bootstrap>", line 693, in _load
  File "<frozen importlib._bootstrap>", line 673, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 697, in exec_module
  File "<frozen importlib._bootstrap>", line 222, in _call_with_frames_removed
  File "....\test_user.py", line 3, in <module>
    from gui.server import create_app
  File "....\server.py", line 2, in <module>
    eventlet.monkey_patch(thread=True)
  File "....\env\lib\site-packages\eventlet\patcher.py", line 265, in monkey_patch
    _green_existing_locks()
  File "....\env\lib\site-packages\eventlet\patcher.py", line 356, in _green_existing_locks
    if isinstance(obj, rlock_type):
ReferenceError: weakly-referenced object no longer exists
```

It seems to be connected to patching the thread module as this works:
```
eventlet.monkey_patch(thread=False)
```

**Comment by frankiedrake** [comment]
I'm not testing my app, I'm just trying to run it in `gunicorn` and I'm getting this error...
My `requirements.txt` is the following:
gunicorn==20.1.0
eventlet==0.30.2
vosk==0.3.43
flask==2.1.3
APScheduler==3.9.1
colorlog==6.6.0
consulate==0.6.0
regex==2022.6.2
soundfile==0.10.3.post1
resemblyzer==0.1.1.dev0
webrtcvad==2.0.10
py-markdown-table==0.2.2
websockets==10.3
psycopg==3.0.16

**Comment by panyogesh** [comment]
Same here...on 3.8.10.. If anyone has seen kindly update for any solution.

**Comment by temoto** [comment]
Possibly fixed in 0.33.1

**Comment by frankiedrake** [comment]
> Possibly fixed in 0.33.1

Seems not... For now I got rid of using requests lib in my app to avoid this error and I tried to update `eventlet` version to 0.33.1, but, using `requests` lib result in stackoverflow error

**Comment by devopstales** [comment]
Still problem with eventlet==0.33.3
